src/handler/mod.rs

```diff
@@ -0,0 +1,117 @@
+use anyhow::{bail, Error, Result};
+use std::fs::OpenOptions;
+use std::io::Write;
+use std::process::Command;
+use std::str;
+
+pub fn handle_reboot(force: bool) -> Result<(), Error> {
+    if !force {
+        match get_boot_counter() {
+            Some(-1) => bail!("boot_counter is -1"),
+            None => bail!("boot_counter is not set"),
+            _ => {}
+        }
+    }
+    log::info!("restarting system");
+    Command::new("systemctl").arg("reboot").status()?;
+    Ok(())
+}
+
+pub fn handle_rollback() -> Result<(), Error> {
+    match get_boot_counter() {
+        Some(-1) => {
+            let status = Command::new("rpm-ostree").arg("rollback").status()?;
+            if status.success() {
+                return Ok(());
+            }
+            bail!("process exited with code {}", status.code().unwrap());
+        }
+        _ => bail!("boot_counter is either unset or not equal to -1"),
+    }
+}
+
+pub fn set_boot_counter(reboot_count: i32) -> Result<()> {
+    match get_boot_counter() {
+        Some(_) => Ok(()),
+        None => {
+            Command::new("grub2-editenv")
+                .arg("-")
+                .arg("set")
```

Comment:
for all `grub2-editenv` commands, the first arg `.arg("-")` is not needed, you can call it as `grub2-editenv X` and that would be it

Reply:
does it work for you? when I try in terminal 
`sudo grub2-editenv set abc=1
Unknown command `abc=1'.
Usage: grub2-editenv [OPTION...] FILENAME COMMAND
Try 'grub2-editenv --help' or 'grub2-editenv --usage' for more information.`

Reply:
yep that's right, I had only tested it with list